pkg/kubectl/cmd/attach.go

```diff
@@ -230,25 +233,30 @@ func (p *AttachOptions) Run() error {
 }
 
 // GetContainer returns the container to attach to, with a fallback.
-func (p *AttachOptions) GetContainer(pod *api.Pod) api.Container {
+func (p *AttachOptions) GetContainer(pod *api.Pod) (*api.Container, error) {
 	if len(p.ContainerName) > 0 {
-		for _, container := range pod.Spec.Containers {
-			if container.Name == p.ContainerName {
-				return container
+		for i := range pod.Spec.Containers {
```

Comment:
Returning `*api.Container` does seem like an improvement, but I think it should return a pointer to a copy, not the value in the slice. That would preserve the original semantics, which might matter.